File: core/support/src/main/java/com/reservation/support/response/ApiResponse.java
package com.reservation.support.response;

public record ApiResponse<T>(
	boolean success,
	T data
) {
	public static <T> ApiResponse<T> ok(T data) {
		return new ApiResponse<>(true, data);
	}

	public static <T> ApiResponse<T> noContent() {
		return new ApiResponse<>(true, null);
	}
}

File: core/support/src/main/kotlin/com/reservation/support/response/ApiResponse.kt
package com.reservation.support.response

@JvmRecord
data class ApiResponse<T>(
    val success: Boolean,
    val data: T
) {
    companion object {
        fun <T> ok(data: T): ApiResponse<T> {
            return ApiResponse(true, data)
        }

        @JvmStatic
        fun <T> noContent(): ApiResponse<T?> {
            return ApiResponse(true, null)
        }
    }
}

